src/providers/aws/dynamodb.ts

```diff
@@ -109,6 +112,56 @@ export default class DynamoDBProvider extends CloudProvider<_Record> {
     return this.opts.shardPollingInterval || 5000;
   }
 
+  // Get or create a shared shard observable
+  private getSharedShardObservable(signal: AbortSignal): Observable<Shard> {
```

Comment:
nit: rename to `private get shards()` as an getter with no setter, and use signal that's already in opts